rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/common/MQRModeDetails.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SonarLint Core - RPC Protocol
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.rpc.protocol.common;
+
+import java.util.List;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.rules.ImpactDto;
+
+public class MQRModeDetails {
+
+  private final CleanCodeAttribute cleanCodeAttribute;
+  private final List<ImpactDto> impacts;
+
+  public MQRModeDetails(CleanCodeAttribute cleanCodeAttribute, List<ImpactDto> impacts) {
```

Comment:
This is `org.sonarsource.sonarlint.core.rpc.protocol.common.CleanCodeAttribute`, which does not contain the clean code category. I think we need to explicitly use `org.sonarsource.sonarlint.core.commons.CleanCodeAttribute`, OR add categories to the one that is used

Reply:
You should be able to use this util class from the client: https://github.com/SonarSource/sonarlint-core/blob/master/client/java-client-utils/src/main/java/org/sonarsource/sonarlint/core/client/utils/CleanCodeAttribute.java#L59